ansible_collections/arista/avd/roles/eos_cli_config_gen/tasks/main.yml

```diff
@@ -14,53 +14,19 @@
   run_once: true
   register: avd_requirements
 
-- name: Create required output directories if not present
+- name: Generate eos intended configuration and device documentation
+  # TODO this loses the capability to only generate documentation
   tags: [build, provision]
-  ansible.builtin.file:
-    path: "{{ item }}"
-    state: directory
-    mode: 0775
-  loop:
-    - "{{ structured_dir }}"
-    - "{{ documentation_dir }}"
-    - "{{ eos_config_dir }}"
-    - "{{ devices_dir }}"
-  delegate_to: localhost
-  run_once: true
-
-- name: Include device intended structure configuration variables
-  tags: [build, provision]
-  ansible.builtin.include_vars:
-    file: "{{ filename }}"
-  delegate_to: localhost
-  when: structured_config is not defined and lookup('first_found', filename, skip=True, errors='ignore')
-  vars:
-    filename: "{{ structured_dir }}/{{ inventory_hostname }}.{{ avd_structured_config_file_format }}"
-
-- name: Generate eos intended configuration
-  tags: [build, provision]
-  arista.avd.validate_and_template:
-    template: eos-intended-config.j2
-    dest: "{{ eos_config_dir }}/{{ inventory_hostname }}.cfg"
-    mode: 0664
-    schema_id: "{{ role_name }}"
-    conversion_mode: "{{ avd_data_conversion_mode }}"
-    validation_mode: "{{ avd_data_validation_mode }}"
-    #cprofile_file: "config-{{inventory_hostname}}.prof"
-  delegate_to: localhost
-  register: eosconfig
-
-- name: Generate device documentation
-  tags: [build, provision, documentation]
-  arista.avd.validate_and_template:
-    template: "eos-device-documentation.j2"
-    dest: "{{ devices_dir }}/{{ inventory_hostname }}.md"
-    mode: 0664
-    schema_id: "{{ role_name }}"
+  arista.avd.eos_cli_config_gen:
+    structured_config_filename: "{{ structured_config_filename }}"
+    config_filename: "{{ eos_config_dir }}/{{ inventory_hostname }}.cfg"
+    documentation_filename: "{{ devices_dir }}/{{ inventory_hostname }}.md"
+    read_structured_config_from_file: "{{ structured_config is not arista.avd.defined and lookup('first_found', structured_config_filename, skip=True, errors='ignore') is not arista.avd.defined([]) }}"
     conversion_mode: "{{ avd_data_conversion_mode }}"
     validation_mode: "{{ avd_data_validation_mode }}"
-    add_md_toc: true
-    md_toc_skip_lines: 3
-    #cprofile_file: "docs-{{inventory_hostname}}.prof"
+    generate_device_doc: "{{ generate_device_documentation | arista.avd.default(true) and 'documentation' not in ansible_skip_tags }}"
```

Comment:
we can add a `generate_device_cfg: "{{ ansible_run_tags is arista.avd.contains([all, build, provision]) and ansible_skip_tags is not arista.avd.contains([build, provision]) }}"` knob to be able to _only_ do the docs?
(Would also require adding `documentation` to the tags on the task.

Reply:
We should probably also handle tags for the documentation in a similar way, to only generate if all or documentation tag is set and documentation is not skipped.

Reply:
[3ca53c1](https://github.com/aristanetworks/avd/pull/4117/commits/3ca53c1d0c6733fae6b0f44719e5d9dbd299b0b6)

but still work in progress it seems

Reply:
> Would also require adding documentation to the tags on the task
This is not working well :) as for playbook with the `--skip-tags=documentation` then we don't run the task and we are sad (and CI as well)
it feels like I almost need to remove tags.. and handle them only in the variables

Reply:
Alternatively we need to keep two tasks running the same plugin. One for cfg and one for doc. Just until 5.0.